src/i_sdlmusic.c

```diff
@@ -645,18 +646,42 @@ static boolean ReadSubstituteConfig(char *filename)
         return false;
     }
 
-//    old_subst_music_len = subst_music_len;
+    length = M_FileLength(fs);
 
-    while (!feof(fs))
+    buffer = malloc(length + 1);
+    if (buffer == NULL)
     {
-        char *retval;
-        M_StringCopy(line, "", sizeof(line));
-        retval = fgets(line, sizeof(line), fs);
+        fclose(fs);
+        I_Error("Out of memory while trying to read music substitution config \"%s\" (%ld bytes)", filename, length);
+    }
+
+    retval = fread(buffer, 1, length, fs);
+    if (retval != length) {
```

Comment:
Please pay attention to coding style; you're using inconsistent braces here compared to the rest of the codebase, and not keeping to the 80 column limit. Read the `HACKING.md` guide if you haven't already.